nameserver/entry.go

```diff
@@ -233,8 +233,8 @@ type GossipData struct {
 }
 
 func (g *GossipData) Merge(o router.GossipData) {
-	checkAndPanic(CaseSensitive(g.Entries))
-	defer func() { checkAndPanic(CaseSensitive(g.Entries)) }()
+	checkAndPanic(CaseInsensitive(g.Entries))
+	defer func() { checkAndPanic(CaseInsensitive(g.Entries)) }()
```

Comment:
Correct.

> We have traditionally preferred to raise an error and drop the connection when bad data is received from another peer, as opposed to panicking.

That would entail adding error returns to a whole bunch of methods. And I don't think that is justifiable here. We could, alternatively, not bother checking.